src/Stratis.Bitcoin.Features.Wallet/WalletRPCController.cs

```diff
@@ -27,20 +33,78 @@ public class WalletRPCController : FeatureController
+        [ActionName("walletpassphrase")]
+        [ActionDescription("Stores the wallet decryption key in memory for the indicated number of seconds. Issuing the walletpassphrase command while the wallet is already unlocked will set a new unlock time that overrides the old one.")]
+        [NoTrace]
+        public bool UnlockWallet(string passphrase, int timeout)
+        {
+            Guard.NotEmpty(passphrase, nameof(passphrase));
+
+            WalletAccountReference account = this.GetAccount();
+
+            // Length of expiry of the unlocking, restricted to max duration.
+            TimeSpan duration = new TimeSpan(0, 0, Math.Min(timeout, maxDurationInSeconds));
+
+            try
+            {
+                this.walletTransactionHandler.CacheSecret(account, passphrase, duration);
+            }
+            catch (SecurityException exception)
+            {
+                throw new RPCServerException(RPCErrorCode.RPC_INVALID_REQUEST, exception.Message);
+            }
+            return true; // NOTE: Have to return a value or else RPC middleware doesn't serialize properly.
+        }
+
+        [ActionName("walletlock")]
+        [ActionDescription("Removes the wallet encryption key from memory, locking the wallet. After calling this method, you will need to call walletpassphrase again before being able to call any methods which require the wallet to be unlocked.")]
+        public bool LockWallet()
         {
             WalletAccountReference account = this.GetAccount();
-            return uint256.Zero;
+            this.walletTransactionHandler.ClearCachedSecret(account);
+            return true; // NOTE: Have to return a value or else RPC middleware doesn't serialize properly.
+        }
+
+        [ActionName("sendtoaddress")]
+        [ActionDescription("Sends money to a bitcoin address. Requires wallet to be unlocked using walletpassphrase.")]
+        public async Task<uint256> SendToAddressAsync(BitcoinAddress address, decimal amount, string commentTx, string commentDest)
+        {
+            WalletAccountReference account = this.GetAccount(); 
+            TransactionBuildContext context = new TransactionBuildContext(this.fullNode.Network)
+            {
+                AccountReference = this.GetAccount(),
+                Recipients = new [] {new Recipient { Amount = Money.Coins(amount), ScriptPubKey = address.ScriptPubKey } }.ToList(),
+                WalletPassword = "_" // Want private key to pull from cache, so pass in non empty string so tx will be signed.
+            };
+
+            try
+            {
+                Transaction transaction = this.walletTransactionHandler.BuildTransaction(context);
+                await this.broadcasterManager.BroadcastTransactionAsync(transaction);
+
+                uint256 hash = transaction.GetHash();
+                return hash;
+            }
+            catch (SecurityException exception)
```

Comment:
Did you plan to have separate RPC server exception here?

Reply:
Good idea. It's updated now with a better error code.